controllers/controlplane_controller.go

```diff
@@ -139,6 +139,52 @@ func (r *ControlPlaneReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		r.Log.Info(fmt.Sprintf("PasswordSecret %s exists", instance.Spec.PasswordSecret))
 	}
 
+	ipsetDetails := common.IPSet{
+		Networks:            instance.Spec.Controller.Networks,
+		Role:                controlplane.Role,
+		HostCount:           controlplane.Count,
+		VIP:                 true,
+		AddToPredictableIPs: false,
+	}
+	ipset, op, err := common.OvercloudipsetCreateOrUpdate(r, instance, ipsetDetails)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if op != controllerutil.OperationResultNone {
+		r.Log.Info(fmt.Sprintf("IPSet for %s successfully reconciled - operation: %s", instance.Name, string(op)))
+		return ctrl.Result{RequeueAfter: time.Second * 10}, nil
```

Comment:
shouldn't the success path be different to the pending replicas case?..

Reply:
Not sure I understand your comment. OperationResultNone [1] means that the resource has not been changed. 
But I think we could remove the reconcile here and just log the done action. It does not hurt, but it seems to be a not necessary reconcile 

[1] https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#OperationResult